lib/WebService/Stripe.pm

```diff
@@ -40,155 +40,181 @@ method next(HashRef $thing, HashRef :$query, HashRef :$headers) {
 }
 
-method add_source(HashRef|Str $cust, HashRef $data, :$headers) {
-    $cust = $cust->{id} if ref $cust;
-    return $self->post( "/v1/customers/$cust/sources", $data, headers => $headers );
+method add_source($customer_id!, $data!, $opts = {}) {
+    return $self->_stripe_req(post => "/v1/customers/$customer_id/sources",
+        $data, $opts);
 }
 
-method create_token(HashRef $data, :$headers) {
-    return $self->post( "/v1/tokens", $data, headers => $headers );
+method create_token($data!, $opts = {}) {
+    return $self->_stripe_req(post => '/v1/tokens', $data, $opts);
 }
 
-method get_token(Str $id, :$headers) {
-    return $self->get( "/v1/tokens/$id", {}, headers => $headers );
+method get_token($token_id!, $query = {}, $opts = {}) {
+    return $self->_stripe_req(get => "/v1/tokens/$token_id", $query, $opts);
 }
 
-method create_account(HashRef $data, :$headers) {
-    return $self->post( "/v1/accounts", $data, headers => $headers );
+method create_account($data!, $opts = {}) {
+    return $self->_stripe_req(post => '/v1/accounts', $data, $opts);
 }
 
-method get_account(Str $id, :$headers) {
-    return $self->get( "/v1/accounts/$id", {}, headers => $headers );
+method get_account($account_id!, $query = {}, $opts = {}) {
+    return $self->_stripe_req(get => "/v1/accounts/$account_id", $query, $opts);
 }
 
-method update_account(Str $id, HashRef :$data!, :$headers) {
-    return $self->post( "/v1/accounts/$id", $data, headers => $headers );
+method update_account($account_id!, $data!, $opts = {}) {
+    return $self->_stripe_req(post => "/v1/accounts/$account_id", $data, $opts);
 }
 
-method upload_identity_document(HashRef|Str $account_id, Str $filepath) {
+method upload_identity_document($data!, $opts = {}) {
+    my $account_id = $data->{'stripe_account'} || $opts->{'stripe_account'}
+        or croak 'data.stripe_account or opts.stripe_account is required';
+    my $filepath = $data->{'filepath'}
+        or croak 'data.filepath is required';
+
     return $self->req(
         POST FILE_UPLOADS_URL,
-        Stripe_Account => ( ref $account_id ? $account_id->{id}: $account_id ),
+        Stripe_Account => "$account_id",
         Content_Type   => 'form-data',
         Content        => [
             purpose => FILE_PURPOSE_ID_DOCUMENT,
-            file    => [ $filepath ],
+            file    => [ "$filepath" ],
         ],
     );
 }
 
-method add_bank(HashRef $data, Str :$account_id!, :$headers) {
-    return $self->post(
-        "/v1/accounts/$account_id/bank_accounts", $data, headers => $headers );
+method add_bank($account_id!, $data!, $opts = {}) {
+    return $self->_stripe_req(post => "/v1/accounts/$account_id/bank_accounts",
+        $data, $opts);
 }
 
-method update_bank(Str $id, Str :$account_id!, HashRef :$data!, :$headers) {
-    return $self->post( "/v1/accounts/$account_id/bank_accounts/$id", $data,
-        headers => $headers );
+method update_bank($account_id!, $bank_id!, $data!, $opts = {}) {
```

Comment:
This is way to many positional params imo, which makes this interface confusing. Someone has to rememeber that the first param is the `$account_id` and the second is the `$bank_id`, which is not intuitive at all imo, and hard to remember. The existing interface is `$stripe->update_bank($bank_id, account_id => ...)`. There is only one positional param to remember, the bank_id, which makes sense, because the method is named `update_bank`.